package.json

```diff
@@ -130,6 +130,14 @@
     "webpack": "^5.44.0",
     "webpack-dev-server": "^3.11.2"
   },
+  "peerDependencies": {
+    "typescript": ">=4.2.0"
+  },
+  "peerDependenciesMeta": {
+    "typescript": {
+      "optional": true
```

Comment:
Does this have anything to do with the pre-installation of peer dependencies in npm 7?

> UPDATE: npm versions 1, 2, and 7 will automatically install peerDependencies if they are not explicitly depended upon higher in the dependency tree. For npm versions 3 through 6, you will receive a warning that the peerDependency is not installed instead.

Reply:
Per NPM:

> When a user installs your package, npm will emit warnings if packages specified in `peerDependencies` are not already installed. The peerDependenciesMeta field serves to provide npm more information on how your peer dependencies are to be used. Specifically, it allows peer dependencies to be marked as optional.
> —[NPM docs](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependenciesmeta)

So `optional` here controls whether to emit the missing peer dependency warning but does not prevent NPM 7 from always installing `typescript`. Looks like we need to research the way to opt-out of that behavior so that JavaScript developers don't get their `node_modules` polluted by the packages they won't use.

Reply:
I can try and run some actual tests on npm 7 in the future, but if I'm reading this correctly and it's still accurate it seems like this should be fine.

> peerOptional deps are deps that are listed in peerDependencies and also have an entry in peerDependenciesMeta containing {"optional": true}. These sorts of dependencies are not installed by default, but if installed, must be resolved to a version that satisfies the optional peer dependency relationship.
https://github.com/npm/rfcs/discussions/289#discussioncomment-131068

Reply:
@willbuckingham, thank you for sharing that excerpt. We can proceed with the `optional` flag and iterate on the developers' feedback once we get some.